highfidelity/hifi
# Add support for ATP protocol to QML

Register a custom QNetworkAccessManager with the QML engine to handle ATP urls.

Fixes [FogBugz 1023](https://highfidelity.fogbugz.com/f/cases/1023/QML-Windows-10)

### Testing
1. Upload a file called "ExampleWindow.qml" to the asset server root folder (see below for contents)
2. Upload an image called "image-asset.jpg" (any JPG image off google should do)
3. Copy-paste the JavaScript below into the console
4. A window should appear with the previously uploaded image in it
5. Try changing the url in ExampleWindow.qml to `/image-asset.jpg`, then to `atp:/image-asset.jpg`, making sure to reupload ExampleWindow.qml and restart interface each time

ExampleWindow.qml:
```
import QtQuick 2.3

Image {
    width: 400;
    height: 150;
    fillMode: Image.PreserveAspectFit;
    source: "image-asset.jpg";
}
```

javascript code:
```
var myWindow = new OverlayWindow({
    title: "My Window",
    source: "atp:/exampleWindow.qml",
    width: 400,
    height: 150,
    visible: true
});
```
